### PR TITLE
adding depends_on functionality between iam_assumegroup and iam_masterassume

### PR DESCRIPTION
**Pull request by bleachbyte**
## TL;DR

The `policy_depends_on` variable in `iam_assumegroup` can be used in conjunction with another value, i.e. the output `policy_arns` in the `iam_masterassume` module, to make `iam_assumegroup` wait for specific IAM policies to exist before attempting to create their respective group attachments.

## Detail

The `iam_assumegroup` module is used to attach IAM policies to groups, and the policies themselves are generated OUTSIDE of this module (and done instead within `iam_masterassume`). Terraform is unable to calculate the proper policy ARNs itself due to this circular dependency, so the `role_group_map` is used with the `master_account_id` to generate the ARN values in the same module.

HOWEVER: If the policies do not already exist before Terraform attempts to create these attachments, the `apply` operation will fail -- even if other resources/modules within it create the policies in parallel. This means that a _subsequent_ `apply` will be successful, but creates too much possibility of orphaned resources / inconsistent state.

This PR solves this dependency loop:

1. Since Terraform does not currently support `depends_on` capabilities on a `module` basis, the variable `policy_depends_on` can be used instead on a per-resource basis, causing the policy attachment creation to wait until _all_ other resources specified by the value(s) in `policy_depends_on` exist (i.e. IAM policy ARNs).
2. To provide a value for `policy_depends_on` the `iam_masterassume` module now provides an output, `policy_arns`, which is a list of the ARNs of all policies created by `iam_masterassume`. This list is not populated until _all_ policies have been created.

### CODE REVIEW

File: iam_masterassume/main.tf [hunk 1]
```diff
@@ -46,3 +46,10 @@ resource "aws_iam_policy" "account_role_policy" {
   description = "Policy to allow user to assume ${split("Assume", each.key)[1]} role in ${split("Assume", each.key)[0]} account(s)."
   policy = data.aws_iam_policy_document.role_policy_doc[each.key].json
 }
+
+# -- Outputs --
+
+output "policy_arns" {
+  description = "Reference this output in order to depend on policy creation being complete."
+  value       = values(aws_iam_policy.account_role_policy)[*]["arn"]
```
Comment (pauldoomgov) [inline iam_masterassume/main.tf:54]: value() with splat?   🕶️  Nice.
